azavea/iow-boundary-tool
# Add Ingested Tile Support, National Municipal Layer

## Overview

We need to display vector datasets that are too large to be loaded into the browser, such as the National Municipal Layer which is a ~56MB GeoJSON file. To render them in a performant manner, we convert the layer into Vector Tiles and host it on S3 + CloudFront, and load it from there using the [Leaflet VectorGrid plugin](https://leaflet.github.io/Leaflet.VectorGrid/vectorgrid-api-docs.html).

This PR adds the scripts to generate the vector tiles, the infrastructure to host it, the wiring to configure it, and the front-end components to render it. It also adds documentation around how to ingest future datasets.

Closes #233 

### Demo

https://github.com/azavea/iow-boundary-tool/assets/1430060/93748a57-1d9c-431f-a7d8-60a690288639

### Notes

~~Since we're using a CloudFront distribution to serve the tiles, it is likely we'll need to add an invalidate mechanism for whenever we publish new tiles to ensure that the latest tiles are always served.~~ Added in 337285f.

The previous Municipal Boundary layer had a name for every boundary that was shown as labels at certain zoom levels. This dataset does not have those labels, so none are shown.

## Testing Instructions

- Check out this branch and `update` and `server`
- Go to http://localhost:4545/ and login as `a1@azavea.com`
- Open a submission, go to its Map view
- Play around with the Municipal Boundary layer
  - [x] Ensure toggling it on and off works
  - [x] Ensure it does not interfere with the Parcel layer and is generally shown in the right spot (above the parcels, beneath the user drawn boundary)
  - [x] Ensure you can see municipal boundaries from around the country

## Checklist
